apps/site/api/views/tags_views.py

```diff
@@ -1,60 +1,16 @@
-from rest_framework import generics, serializers
-from tagging.models import Tag, TaggedItem
-from localground.apps.site import models
-from localground.apps.lib.helpers.generic import FastPaginator
-from rest_framework.permissions import AllowAny
-from localground.apps.site.api.views import *
-
-
-class TagSerializer(serializers.Serializer):
-    id = serializers.SerializerMethodField('get_id')
-    name = serializers.SerializerMethodField('get_name')
-    count = serializers.SerializerMethodField('get_count')
-
-    def get_id(self, obj):
-        try:
-            return obj.id
-        except Exception:
-            return obj.get('tag__id')
-
-    def get_name(self, obj):
-        try:
-            return obj.name
-        except Exception:
-            return obj.get('tag__name')
-
-    def get_count(self, obj):
-        try:
-            return obj.count
-        except Exception:
-            return obj.get('tag__count')
-
-
-class TagList(generics.ListAPIView):
-    # http://django-tagging.googlecode.com/svn/trunk/docs/overview.txt
-    serializer_class = TagSerializer
-    paginate_by = 100
-    paginator_class = FastPaginator
-
-    def get_queryset(self):
-        if self.kwargs.get('model_name_plural') is None:
-            from django.db.models import Count
-            return (
-                TaggedItem.objects
-                .values('tag__name')
-                .annotate(Count('tag'))
-                .order_by('-tag__count')
-            )
-        else:
-            model_class = models.Base.get_model(
-                model_name_plural=self.kwargs.get('model_name_plural')
-            )
-            queryset = None
-            if self.request.user.is_authenticated():
-                queryset = model_class.objects.get_objects(self.request.user)
-            else:
-                queryset = model_class.objects.get_objects_public(
-                    access_key=self.request.GET.get('access_key')
-                )
-            return Tag.objects.usage_for_queryset(queryset, counts=True)
-            # return Tag.objects.cloud_for_model(model_class)
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from localground.apps.site.models import Audio, Photo, Print, Video
+from django.db.models import Func, F
+
+class TagList(APIView):
```

Comment:
I'd prefer not to punt on tests because they don't take long to make and
they're really helpful to flag regressions. I didn't realize this endpoint
wasn't done. If you're still working on this PR and don't want to merge it
yet, let me know, but I'd prefer not to roll new functionality into master
w/o tests. You could also just start by making a stub that (a) inserts one
tagged object into the database for each data type (photo, audio, and
map_image), (b) queries the endpoint, and (c) ensures that there are at
least n tags returned in the array. It's not exhaustive, but it will still
be a really useful indicator to know whether or not autocomplete is
up-and-running, or down.

On Thu, Feb 25, 2016 at 8:27 AM, tap2k <notifications@github.com> wrote:

> In apps/site/api/views/tags_views.py
> <https://github.com/LocalGround/localground/pull/58#discussion_r54121786>:
>
> > -            queryset = None
> > -            if self.request.user.is_authenticated():
> > -                queryset = model_class.objects.get_objects(self.request.user)
> > -            else:
> > -                queryset = model_class.objects.get_objects_public(
> > -                    access_key=self.request.GET.get('access_key')
> > -                )
> > -            return Tag.objects.usage_for_queryset(queryset, counts=True)
> > -            # return Tag.objects.cloud_for_model(model_class)
> > +from rest_framework import status
> > +from rest_framework.views import APIView
> > +from rest_framework.response import Response
> > +from localground.apps.site.models import Audio, Photo, Print, Video
> > +from django.db.models import Func, F
> > +
> > +class TagList(APIView):
>
> Ill need to think about the test for this as it is a property of the
> global state of the app. Anyway can we punt on this until I finish the
> endpoint?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LocalGround/localground/pull/58/files#r54121786>.
>